### DISCUSSION TITLE
[Q]: move from `vendor/bin/rr get` to `COPY --from=spiralscout/roadrunner:2.7.6 /usr/bin/rr /usr/bin/rr` ?

### DISCUSSION THREAD

**Discussion by Ocramius** [roadrunner-server/roadrunner#990] [Q&A] [answered]
### I have an idea!

While running `vendor/bin/rr get`, I often incur into `403` errors due to rate limits (not authenticated against the github API).

This problem is even starker when setting up roadrunner inside a `Dockerfile` (and its related build).

Is copying the binary (from the correct arch) perhaps a better approach?

```Dockerfile
COPY --from=spiralscout/roadrunner:2.7.6 /usr/bin/rr /usr/bin/rr
```

In practice, the above is:

 * cacheable
 * cache is invalidated correctly when version changes (compared to the always cached `rr get`)
 * compatible with @dependabot and @renovatebot (upgrades will be performed automatically)
 * removes the need for HTTP requests performed from within the build context

Thoughts? I know this is mostly for `docker` users, but it seems like a much simpler installation method, and it would allow the deprecation of the `GetBinaryCommand`, which obviously has limits inherent from the GitHub API.

**Comment by rustatian** [comment, marked as answer]
Hey @Ocramius 👋🏻 Yeah, getting RR from the docker is a good option, many users already apply this approach. Also, would be an option is to use some env variable (`GH_TOKEN` for example for the `rr get`) to raise the limit up to 5000 req/h. This is not implemented yet, I'll discuss that with our PHP team. [Velox](https://github.com/roadrunner-server/velox) for example uses `github_token` option to raise the limit.

**Reply by rustatian** [reply]
Also, we have a docker GitHub package if, for some reason, Docker API doesn't respond: https://github.com/roadrunner-server/roadrunner/pkgs/container/roadrunner

**Reply by Ocramius** [reply]
So, the main question is mostly whether we can endorse that in the `README.md` then, and whether `GetBinaryCommand` can be deprecated (after adoption shifts)

**Reply by rustatian** [reply]
For sure we can point attention to that in the `README.md` (would appreciate for the PR). Not sure about `GetBinaryCommand` deprecation as it's just one of the binary delivery methods.

**Reply by rustatian** [reply]
@Ocramius I've updated the `README.md`: https://github.com/roadrunner-server/roadrunner#installation. Thank you for the advice 👍🏻

**Reply by Ocramius** [reply]
Awesome, thank you so much! 

Ref: https://github.com/roadrunner-server/roadrunner/commit/ec99bae2a622f9622209f738df6251b3cb5f4e76
Ref: https://github.com/roadrunner-server/roadrunner/commit/1e62f13eb4cd58f8fbe7f1b44e468a1e089a35d5
Ref: https://github.com/roadrunner-server/roadrunner/commit/86c1843a65c5d83977c5db8b0959932c8c1c2b3c

**Comment by rustatian** [comment]
cc @butschster
